packages/graph/hash_graph/bin/hash_graph/src/args.rs

```diff
@@ -29,14 +29,14 @@ pub struct Args {
     /// - be in the standard format accepted by Rust's `regex` crate.
     ///
     /// - contain a capture group named "shortname" to identify a user's shortname, e.g.
-    ///   `(?P<shortname>[\w]*)`
+    ///   `(?P<shortname>[\w|-]+)`
     ///
     /// - contain a capture group named "kind" to identify the slug of the kind of ontology type
     ///   being hosted (data-type, property-type, entity-type, link-type), e.g.
     ///   `(?P<kind>(?:data-type)|(?:property-type)|(?:entity-type)|(?:link-type))`
     #[clap(
         long,
-        default_value_t = Regex::new(r"http://localhost:3000/@(?P<shortname>\w+)/types/(?P<kind>(?:data-type)|(?:property-type)|(?:entity-type)|(?:link-type))/(?:\w|-)+/").unwrap(),
+        default_value_t = Regex::new(r"http://localhost:3000/@(?P<shortname>[\w|-]+)/types/(?P<kind>(?:data-type)|(?:property-type)|(?:entity-type)|(?:link-type))/(?:\w|-)+/").unwrap(),
```

Comment:
Though I think that having `|` is actually wrong here, it would match against the character which isn't valid in the workspace shortname validation

Reply:
Oops that's a good spot!
```suggestion
        default_value_t = Regex::new(r"http://localhost:3000/@(?P<shortname>(?:\w|-)+)/types/(?P<kind>(?:data-type)|(?:property-type)|(?:entity-type)|(?:link-type))/(?:\w|-)+/").unwrap(),
```

Reply:
Although on that note I wonder if 
`[\w-]+` is better than `(?:\w|-)+`

Reply:
I think we don't want the `(?:\w|-)` grouped like this. It would create a non-capture group for each `\w`, wouldn't it? Perhaps `(?:[\w-]+)` would be better

Reply:
Is that a problem though? It's a _non_-capture group

regex101.com also isn't showing anything strange happening here but I'm not suggesting we can rely on it.
<img width="347" alt="image" src="https://user-images.githubusercontent.com/25749103/189129903-fd3d6a2f-7126-4c9c-8ce1-5cd654d4f98c.png">

I'll probably swap to `[\w-]` anyway but I find that syntax a lot less clear because you don't have the explicit _or_

Reply:
```suggestion
        default_value_t = Regex::new(r"http://localhost:3000/@(?P<shortname>[\w-]+)/types/(?P<kind>(?:data-type)|(?:property-type)|(?:entity-type)|(?:link-type))/[\w-]+/").unwrap(),
```

<img width="1662" alt="image" src="https://user-images.githubusercontent.com/25749103/189130520-d8cd4477-f30a-403e-acc0-8edae38073de.png">

Reply:
Ah no worries then. Up to you, either works for me if it doesn't mean it's wrong :)

Reply:
I tried to google about efficiency and didn't get a clear answer but it is possible that the character classes are more efficient.